scene/2d/path_2d.cpp

```diff
@@ -319,8 +319,24 @@ void PathFollow2D::_bind_methods() {
 void PathFollow2D::set_offset(float p_offset) {
 
 	offset = p_offset;
-	if (path)
+	if (path) {
+		if (path->get_curve().is_valid() && path->get_curve()->get_baked_length()) {
+			float path_length = path->get_curve()->get_baked_length();
+
+			if (loop) {
+				while (offset > path_length)
+					offset -= path_length;
+
+				while (offset < 0)
+					offset += path_length;
```

Comment:
I think this is equivalent to `fposmod(offset, path_length)`.

Reply:
Yep, although it might wrap 1 to 0, which was the problem solved by #34012